### PR TITLE
#24 HistoryController created

### LINKED ISSUE #24: Create History Controller
Create a controller that uses the methods of the HistoryService
Create userPositionDTO and configure xjc

### PR DESCRIPTION
**Pull request by iasatan**
userPositionDTO and UserPositionDTOConverter created
IT tests renamed to end with IT, surfire configured
Resolve #24 issue

### CODE REVIEW

File: controller/src/main/java/uni/miskolc/ips/ilona/tracking/controller/history/HistoryController.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    private HistoryService historyService;
+
+    @Autowired
+    public HistoryController(HistoryService historyService) {
+        this.historyService = historyService;
+    }
+
+    @ResponseStatus(value = HttpStatus.CREATED)
+    @RequestMapping(value = "add", method = RequestMethod.GET, consumes = MediaType.APPLICATION_JSON_VALUE)
+    public void addHistory(@RequestBody UserPositionDTO userPositionDTO) throws ServiceGeneralErrorException, UserNotFoundException {
```
Comment (judit0310) [inline controller/src/main/java/uni/miskolc/ips/ilona/tracking/controller/history/HistoryController.java]: When will it throw ServiceGeneralErrorException?

File: controller/src/main/java/uni/miskolc/ips/ilona/tracking/controller/history/HistoryController.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        UserPosition userPosition = UserPositionDTOConverter.convertToUserPosition(userPositionDTO);
+        try {
+            historyService.addHistory(userPosition.getPosition(), userPosition.getTime(), userPosition.getUser().getUserid());
+        } catch (UserNotFoundException e) {
```
Comment (judit0310) [inline controller/src/main/java/uni/miskolc/ips/ilona/tracking/controller/history/HistoryController.java]: You just rethrow the same exception after logging?

Comment (iasatan) [reply controller/src/main/java/uni/miskolc/ips/ilona/tracking/controller/history/HistoryController.java]: Yes. In this way I can log the user ID and create an exception handler too

File: controller/src/main/java/uni/miskolc/ips/ilona/tracking/controller/history/HistoryController.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            LOG.error("No user found with id: " + userPosition.getUser().getUserid());
+            throw new UserNotFoundException();
+        }
+        LOG.info("History added to user: " + userPosition.getUser().getUserid());
+    }
+
+    @RequestMapping(value = "listHistoryByUserId/{userId}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public List<UserPositionDTO> listHistoryByUser(@PathVariable("userId") String userId) throws DatatypeConfigurationException, UserNotFoundException {
```
Comment (judit0310) [inline controller/src/main/java/uni/miskolc/ips/ilona/tracking/controller/history/HistoryController.java]: DatatypeConfigurationException is not logged like UsernotFound?

File: controller/src/main/java/uni/miskolc/ips/ilona/tracking/controller/history/HistoryController.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        List<UserPositionDTO> userPositionDTOS = null;
```
Comment (judit0310) [inline controller/src/main/java/uni/miskolc/ips/ilona/tracking/controller/history/HistoryController.java]: Are you really want to cause possible NullPointerException?

**Comment by iasatan** [comment]
updated

**Pull request merged**